pkg/river/vm/constant.go

```diff
@@ -18,11 +18,27 @@ func valueFromLiteral(lit string, tok token.Token) (value.Value, error) {
 		return value.Null, nil
 
 	case token.NUMBER:
-		v, err := strconv.ParseInt(lit, 0, 64)
-		if err != nil {
-			return value.Null, err
+		int64Val, err1 := strconv.ParseInt(lit, 0, 64)
+		if err1 == nil {
+			intVal := int(int64Val)
+			if int64Val == int64(intVal) {
+				return value.Int(intVal), nil
+			} else {
+				return value.Int64(int64Val), nil
+			}
```

Comment:
Since we're always going to change the River representation to be either an int64/uint64/float64, you can remove this conditional here to speed things up a little.